model/auth_config.go

```diff
@@ -11,4 +11,17 @@ type AuthConfig struct {
 	AllowedIdentities []client.Identity `json:"allowedIdentities"`
 	GithubConfig      GithubConfig      `json:"githubConfig"`
 	ShibbolethConfig  ShibbolethConfig  `json:"shibbolethConfig"`
+	LdapConfig        LdapConfig        `json:"ldapConfig"`
 }
+
+type TestAuthConfig struct {
+	AuthConfig AuthConfig `json:"authConfig"`
+	Code       string     `json:"code"`
+}
+
+const (
+	AccessModeSetting        = "api.auth.access.mode"
+	AllowedIdentitiesSetting = "api.auth.allowed.identities"
+)
+
+var RancherClient *client.RancherClient
```

Comment:
ad_provider should have all the settings provided to it in the authconfig during init/loadConfig. You should read the settings from there.